hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -352,7 +352,7 @@ public static int mapRecordKeyToFileGroupIndex(String recordKey, int numFileGrou
     }
 
     HoodieTableFileSystemView fsView = new HoodieTableFileSystemView(metaClient, timeline);
-    return fsView.getLatestFileSlices(partition).sorted((s1, s2) -> s1.getFileId().compareTo(s2.getFileId()))
+    return fsView.getLatestUnCompactedFileSlices(partition).sorted((s1, s2) -> s1.getFileId().compareTo(s2.getFileId()))
```

Comment:
With metadata table we begin without any base files. Can this handle the fact that there is no previous file slice during compaction?

Reply:
should be. even today, with bootstrap delta commit in metadata table, we have one log file in the first file slice and there is no pending compaction for the latest file slice.